custom/enikshay/nikshay_datamigration/management/commands/create_enikshay_cases.py

```diff
@@ -0,0 +1,210 @@
+from django.core.management import BaseCommand
+from django.db import transaction
+
+from casexml.apps.case.const import CASE_INDEX_EXTENSION
+from casexml.apps.case.mock import CaseFactory, CaseStructure, CaseIndex
+from corehq.apps.locations.models import SQLLocation
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+
+from custom.enikshay.nikshay_datamigration.models import PatientDetail, Outcome, Followup
+from dimagi.utils.decorators.memoized import memoized
+
+
+class EnikshayCaseFactory(object):
+
+    domain = None
+    patient_detail = None
+
+    def __init__(self, domain, patient_detail):
+        self.domain = domain
+        self.patient_detail = patient_detail
+        self.factory = CaseFactory(domain=domain)
+        self.case_accessor = CaseAccessors(domain)
+
+    @transaction.atomic
+    def create_cases(self):
+        self.create_person_case()
+        self.create_occurrence_cases()
+        self.create_episode_cases()
+        self.create_test_cases()
+
+    def create_person_case(self):
+        self.factory.create_or_update_case(self.person())
+
+    def create_occurrence_cases(self):
+        occurrences = [self.occurrence(outcome) for outcome in self._outcomes]
+        cases = self.factory.create_or_update_cases(occurrences)
+        for occurrence_structure, occurrence_case in zip(occurrences, cases):
+            occurrence_structure.case_id = occurrence_case.case_id
+
+    def create_episode_cases(self):
+        episodes = [self.episode(outcome) for outcome in self._outcomes]
+        cases = self.factory.create_or_update_cases(episodes)
+        for episode_structure, episode_case in zip(episodes, cases):
+            episode_structure.case_id = episode_case.case_id
+
+    def create_test_cases(self):
+        tests = [
+            self.test(followup) for followup in self._followups
+            if Outcome.objects.filter(PatientId=followup.PatientID).exists()
+            # how many followup's do not have a corresponding outcome? how should we handle this situation?
+        ]
+        self.factory.create_or_update_cases(tests)
+
+    @memoized
+    def person(self):
+        return CaseStructure(
+            case_id=self.patient_detail.PregId,
```

Comment:
I think we want to set the nikshay id as a case property (`nikshay_id`), and keep the case_id to be a uuid4. We might also want to set the nikshay id as the `external_id` property, as that is easily indexed.